client.go

```diff
@@ -368,15 +368,15 @@ func (c *Client) receiveLoop() error {
 			sid := streamID(msg.header.StreamID)
 			s := c.getStream(sid)
 			if s == nil {
-				logrus.WithField("stream", sid).Errorf("ttrpc: received message on inactive stream")
+				log.G(c.ctx).WithField("stream", sid).Error("ttrpc: received message on inactive stream")
```

Comment:
These contexts embedded in the struct seemed awkward; there's probably a reason for this, or it was written long time ago; wondering if someone knows the background and if this is something that needs to be changed.

(I didn't look into history yet)